[react-error-overlay] Add an Optional Runtime Option Style iFrame with a Class

dav-is opened the pull request:
To be more [CSP](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) compliant, inline styles should be avoided. I added an alternative to using inline styles on the iframe itself so it doesn't violate CSP.

![Screenshot of change](https://user-images.githubusercontent.com/11590024/48155067-ca7a1800-e28f-11e8-93d2-85dd63b04962.png)
(Tested in Next.js)

mrmckeb commented:
Thanks for this PR, @dav-is. Are you able to put together a quick demonstration project with this in place?

dav-is commented:
@mrmckeb I implemented it inside of next.js so it'd be hard to load all the changes locally, but basically I copied the styles that are added to the iframe `<iframe styles=` and moved it to `<iframe class=` if the option is set. The corresponding CSS becomes the app's responsibility and is inserted in a stylesheet from the server in dev mode.

https://github.com/dav-is/next.js/blob/d1017f8f84579169182ae9558cd9d25c8e496397/packages/next/pages/_document.js#L133
https://github.com/dav-is/next.js/blob/d1017f8f84579169182ae9558cd9d25c8e496397/packages/next/client/dev-error-overlay/hot-dev-client.js#L74

mrmckeb commented:
Sorry @dav-is, I missed your response here (lost in emails). It would be great to get a demo of this to test before merging, if you could put one together?

This is an area of CRA I'm not intimately familiar with, so that would help in testing.
